test/WebJobs.Script.Tests.Integration/TestFunctionHost.cs

```diff
@@ -165,18 +165,31 @@ public async Task RestartAsync(CancellationToken cancellationToken)
             await _hostService.RestartHostAsync(cancellationToken);
         }
 
-        private async Task StartAsync()
+        private Task StartAsync()
         {
-            bool running = false;
-            while (!running)
+            var startTask = Task.Run(async () =>
             {
-                running = await IsHostStarted(HttpClient);
-
-                if (!running)
+                bool running = false;
+                while (!running)
                 {
-                    await Task.Delay(50);
+                    running = await IsHostStarted();
+
+                    if (!running)
+                    {
+                        await Task.Delay(50);
+                    }
                 }
+            });
+
+            if (startTask.Wait(TimeSpan.FromMinutes(1)))
```

Comment:
Added a timeout here, as this is the part which hung forever. I think if the Host doesn't start in a minute, it's safe to fail.